src/libs/cursor.rs

```diff
@@ -256,6 +256,17 @@ where
         self.inc_bytes(1).map(|_| byte)
     }
 
+    fn read_i8(&mut self) -> Result<i8, CodeEofError> {
+        if self.is_eof() {
+            return Err(CodeEofError);
+        }
+        let pos = self.byte_pos as usize;
+        let mut buf = [0u8];
+        buf.copy_from_slice(&self.as_ref()[pos..pos + 1]);
+        let byte = i8::from_le_bytes(buf);
```

Comment:
Can be simplified as

```suggestion
        let byte = self.as_ref()[self.byte_pos as usize] as i8;
```